src/main/services/CaseService.ts

```diff
@@ -67,3 +70,61 @@ export const getPreloginCaseData = (redisClient: RedisClient, guid: string): Pro
     });
   });
 };
+
+export const formatCaseData = (fromApiCaseData: initiateCaseDraftResponse): CaseWithId => ({
+  id: fromApiCaseData.id,
+  state: fromApiCaseData.state,
+  isASingleClaim: fromApiCaseData.case_data.case_type === 'Single' ? YesOrNo.YES : YesOrNo.NO,
+});
+
+export class CaseApi {
+  constructor(private readonly axio: AxiosInstance) {}
+
+  public async getCase(caseTypeId = CcdDataModel.SINGLE_CASE_ENGLAND): Promise<CaseWithId | false> {
+    try {
+      const fetchedCases = await this.getDraftCases(caseTypeId);
+      switch (fetchedCases.length) {
+        case 0: {
+          return false;
+        }
+        default: {
+          return formatCaseData(fetchedCases[fetchedCases.length - 1]);
+        }
+      }
+    } catch (err) {
+      throw new Error('Case could not be retrieved.');
+    }
+  }
+
+  private async getDraftCases(caseTypeId: string): Promise<initiateCaseDraftResponse[]> {
+    try {
+      const response = await this.axio.get(`/caseTypes/${caseTypeId}/cases`, {
+        data: {
+          match: { state: 'Draft' },
+        },
+      });
+      return response.data;
+    } catch (error) {
+      if (error.response) {
+        throw new Error('Cases could not be retrieved, status: ' + error.response.status);
+      } else if (error.request) {
+        throw new Error('Cases not retrieved, request made but no reponse received');
+      } else {
+        throw new Error('Cases not retrieved, something happened in setting up the request');
+      }
+    }
+  }
+}
+
+export const getCaseApi = (userDetails: UserDetails): CaseApi => {
```

Comment:
Should we also use this in the existing createCase function? just so there won't be any duplicate code.

Reply:
Yes we could probably move some of the existing stand alone functions and make them methods in the caseapi class, to avoid duplication of setting up the requests with axios.